File: problems/1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            rest = target - nums[i]
            idx = map.get(nums[i])
            map[rest] = i
            if idx != None  and idx != i:
                return [idx, i]
                

File: problems/1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map: Map<number, number> = new Map();

    for (let i = 0; i < nums.length; i++) {
        const rest = target - nums[i];
        const idx = map.get(nums[i]);
        map.set(rest, i);
        if (idx != undefined && idx != i) {
            return [idx, i];
        }
    }
};